pkg/ecdsa/dkg/message.go

```diff
@@ -0,0 +1,114 @@
+package dkg
+
+import (
+	"fmt"
+	"github.com/keep-network/keep-core/pkg/beacon/gjkr/gen/pb"
+	"github.com/keep-network/keep-core/pkg/crypto/ephemeral"
+	"github.com/keep-network/keep-core/pkg/protocol/group"
+)
+
+const messageTypePrefix = "ecdsa_dkg/"
+
+// ephemeralPublicKeyMessage is a message payload that carries the sender's
+// ephemeral public keys generated for all other group members.
+//
+// The receiver performs ECDH on a sender's ephemeral public key intended for
+// the receiver and on the receiver's private ephemeral key, creating a symmetric
+// key used for encrypting a conversation between the sender and the receiver.
+type ephemeralPublicKeyMessage struct {
+	senderID group.MemberIndex
+
+	ephemeralPublicKeys map[group.MemberIndex]*ephemeral.PublicKey
+}
+
+// SenderID returns protocol-level identifier of the message sender.
+func (epkm *ephemeralPublicKeyMessage) SenderID() group.MemberIndex {
+	return epkm.senderID
+}
+
+// Type returns a string describing an EphemeralPublicKeyMessage type for
+// marshaling purposes.
+func (epkm *ephemeralPublicKeyMessage) Type() string {
+	return messageTypePrefix + "ephemeral_public_key"
+}
+
+// Marshal converts this EphemeralPublicKeyMessage to a byte array suitable for
+// network communication.
+func (epkm *ephemeralPublicKeyMessage) Marshal() ([]byte, error) {
+	ephemeralPublicKeys, err := marshalPublicKeyMap(epkm.ephemeralPublicKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	return (&pb.EphemeralPublicKey{
+		SenderID:            uint32(epkm.senderID),
+		EphemeralPublicKeys: ephemeralPublicKeys,
+	}).Marshal()
+}
```

Comment:
I will think about that in the following PR #3112 since it's where new messages will be added. Personally, I would rather keep everything message-related in a single file as doing a lot of jumps between files is annoying. Also, if we do a single `marshaling.go` file, this is not obvious that only message marshaling code lives there. Anyway, I'm not fixed on this single-file approach and I'm ok with splitting.

Reply:
Cool. Whatever we decide, let's make it consistent with the beacon side. This is all about convention, personal preferences will for sure be different in the team.

Reply:
Sure thing!